unit_tests/test_StaticSVD.cpp

```diff
@@ -95,7 +108,397 @@ int main(int argc, char* argv[])
+                  row_offset,
+                  1,
+                  MPI_INT,
+                  MPI_COMM_WORLD);
+
+    for (int i = d_num_procs - 1; i >= 0; i--) {
+        row_offset[i] = row_offset[i + 1] - row_offset[i];
+    }
+
+    double* samples = new double[15] {0.5377, -1.3077, -1.3499,
+                                      1.8339, -0.4336, 3.0349,
+                                      -2.2588, 0.3426, 0.7254,
+                                      0.8622, 3.5784, -0.0631,
+                                      0.3188, 2.7694, 0.7147};
+
+    double* basis_right_true_ans = new double[15] {
+        3.08158946098238906153E-01,	    -9.49897947980619661301E-02,	-4.50691774108525788911E-01,	
+        -1.43697905723455976457E-01,	9.53289043424090820622E-01,	    8.77767692937209131898E-02,	
+        -2.23655845793717528158E-02,	-2.10628953513210204207E-01,	8.42235962392685943989E-01,	
+        -7.29903965154318323805E-01,	-1.90917141788945754488E-01,	-2.77280930877637610266E-01,	
+        -5.92561353877168350834E-01,	-3.74570084880578441089E-02,	5.40928141934190823137E-02,	
+    };
+
+    double* basis_true_ans = new double[9] {
+        -1.78651649346571794741E-01,	5.44387957786310106023E-01,	    -8.19588518467042281834E-01,	
+        -9.49719639253861602768E-01,	-3.13100149275943651084E-01,	-9.50441422536040881122E-04,	
+        -2.57130696341890396805E-01,	7.78209514167382598870E-01,	    5.72951792961765460355E-01,
+    };
+
+    double* sv_true_ans = new double[3] {
+        4.84486375065219387892E+00,     3.66719976398777269821E+00,     2.69114625366671811335E+00,
+    };
+
+    SLPK_Matrix *d_samples = new SLPK_Matrix;
+    std::vector<int> d_dims;
+    d_dims.resize(static_cast<unsigned>(d_num_procs));
+    MPI_Allgather(&d_num_rows, 1, MPI_INT, d_dims.data(), 1, MPI_INT, MPI_COMM_WORLD);
+    int d_total_dim = 0;
+    std::vector<int> d_istarts(static_cast<unsigned>(d_num_procs), 0);
+
+    for (unsigned i = 0; i < static_cast<unsigned>(d_num_procs); ++i) {
+        d_total_dim += d_dims[i];
+        if (i > 0) {
+            d_istarts[i] = d_istarts[i-1] + d_dims[i-1];
+        }
+    }
+
+    /* TODO: Try doing this more intelligently and see if it makes a difference */
+    int d_nprow = d_num_procs;
+    int d_npcol = 1;
+    int d_blocksize = d_total_dim / d_nprow;
+    if (d_total_dim % d_nprow != 0) {
+        d_blocksize += 1;
+    }
+    initialize_matrix(d_samples, d_total_dim, num_total_cols,
+                      d_nprow, d_npcol, d_blocksize, d_blocksize);
+
+    int d_num_samples = 0;
+    for (int s = 0; s < 5; s++)
+    {
+        const double *u_in = &samples[3*s];
+        for (int rank = 0; rank < d_num_procs; ++rank) {
+            scatter_block(d_samples, d_istarts[static_cast<unsigned>(rank)]+1,
+                        s+1, u_in, d_dims[static_cast<unsigned>(rank)],
+                        1, rank);
+        }
+    }
+    printf("original\n");
+    MPI_Barrier(MPI_COMM_WORLD);
```

Comment:
Are these barriers necessary? If so, should they be in some function?

Reply:
`MPI_Barrier` was just for debugging, which does not do anything. Removed now.